src/Core/src/Platform/Android/TimePickerExtensions.cs

```diff
@@ -0,0 +1,30 @@
+﻿using Android.Text.Format;
+
+namespace Microsoft.Maui
+{
+	public static class TimePickerExtensions
+	{
+		public static void UpdateFormat(this MauiTimePicker mauiTimePicker, ITimePicker view)
+		{
+			mauiTimePicker.SetTime(view);
+		}
+
+		public static void UpdateTime(this MauiTimePicker mauiTimePicker, ITimePicker view)
+		{
+			mauiTimePicker.SetTime(view);
+		}
+
+		internal static void SetTime(this MauiTimePicker mauiTimePicker, ITimePicker timePicker)
+		{
+			var time = timePicker.Time;
+			var format = timePicker.Format;
+
+			mauiTimePicker.Text = time.ToFormattedString(format);
+		}
+
+		internal static bool Is24HourView(this MauiTimePicker mauiTimePicker, ITimePicker? timePicker)
```

Comment:
nit: this logic flows back and forth between things. Maybe this is better to be an extension method on ITimePicker that takes a param of Context? Even though it is now used for the MauiTimePIcker, it really is not about that. It _uses_ the context from the picker, so maybe this:

```csharp
internal static bool Is24HourView(this ITimePicker timePicker, Context context);
```

Reply:
I think this is only useful in the context of the handler itself, and it requires both the virtual view and the native view to make this decision. So it should just be a private method on the handler, not an extension.